client/scripts/Player.js

```diff
@@ -0,0 +1,284 @@
+class Player {
+	constructor() {
+		this.isHost = false
+		this.isFakeArtist = false
+		this.socket = io()
+		this.listOfPlayers = null
+		this.room_id = null
+		this.color = null
+		this.number = null
+		this.pad = null
+		this.board = null
+		this.target = document.getElementById("main")
+		this.isGameStarted = false
+		this.clue = null
+	}
+	// action handlers after receiving messages from server:
+	socketInit() {
+		// redirect to homepage on server connection rejection
+		this.socket.on("disconnect", reason => {
+			console.log(reason)
+			alert("disconnected")
+			window.location = "/"
+		})
+		// redirect to homepage on other connection errors
+		this.socket.on("connect_error", error => {
+			console.log(error)
+			alert("connection error")
+			window.location = "/"
+		})
+		// binding the reference of 'this' to the class
+		// else the callback methods will reference the Socket once called
+		this.socket.on("start_game", this.startGame.bind(this))
+		this.socket.on("room_id", this.displayRoomID.bind(this))
+		this.socket.on("new_image", this.updateImageFromServer.bind(this))
+		this.socket.on("player_number", this.assignNumber.bind(this))
+		this.socket.on("enable_turn", this.enableActions.bind(this))
+		this.socket.on("player_list", this.listPlayersOnPage.bind(this))
+		this.socket.on("assigned_as_host", this.becomeHost.bind(this))
+		this.socket.on("room_clue", this.assignClue.bind(this))
+		this.socket.on("current_player", this.displayCurrentPlayer.bind(this))
+		this.socket.on("assigned_fake_artist", this.setToFakeArtist.bind(this))
+	}
+	setToFakeArtist() {
+		this.isFakeArtist = true
+	}
+	displayRoomClue() {
+		if (this.isFakeArtist) {
+			this.target.querySelector("#room_clue").innerHTML =
+				"You are the fake artist"
+		} else {
+			this.target.querySelector(
+				"#room_clue"
+			).innerHTML = `The Word is ${this.clue}`
+		}
+	}
+	displayCurrentPlayer(player_id) {
+		const playerName = player_id
+		if (playerName == this.socket.id) {
+			playerName == "Your"
+		}
+		this.target.querySelector(
+			"#current_player"
+		).innerHTML = `It is ${playerName}'s turn`
+	}
+	assignClue(clue) {
+		this.clue = clue
+	}
+	becomeHost() {
+		this.isHost = true
+		if (!this.isGameStarted) {
+			this.target.append(this.createStartGameButton())
+		}
+	}
+	listPlayersOnPage(list_of_players) {
+		const target = document.querySelector("#list_of_players")
+		while (target.firstChild) {
+			target.firstChild.remove()
+		}
+		list_of_players.forEach((player, index) => {
+			const node = document.createElement("div")
+			node.innerHTML = `${index + 1}. ${player}`
+			if (player == this.socket.id) {
+				node.style.fontWeight = "bold"
+			}
+			target.append(node)
+		})
+	}
+	assignNumber(assigned_number) {
```

Comment:
conventional naming for functions liek this is
`setAssignedNumber(assigned_number)`